tests/pytorch_tests/trainable_infrastructure_tests/trainable_pytorch/test_pytorch_base_quantizer.py

```diff
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
+from typing import List, Any
```

Comment:
A general comment about the tests - don't you think we should add some tests for the freeze functionality not just through the QAT/GPTQ, but straight on the infrastructure?

Reply:
In principle yes. In practice I'm not sure what level of infrastructure you refer to. Base quantizer doesn't really do anything except having the flag with the default value. Only ste activation quantizers actually have the interface for and implement the freezing, as we talked. I think we agreed this is not ideal but it would require some refactoring to add this at infrastructure level and we don't really expect any additional quantizers to need this in the near future.
I would test the quantizers directly, but I didn't see any such existing tests, qat/gptq was the closest I found. My understanding was that we don't want to rewrite/write any new tests for this task. Please correct me if I missed or misunderstood anything.

Reply:
Actually it seems there is not a single test that uses UniformQuantizer, so it is not tested even indirectly. I will add something

Reply:
> In principle yes. In practice I'm not sure what level of infrastructure you refer to. Base quantizer doesn't really do anything except having the flag with the default value. Only ste activation quantizers actually have the interface for and implement the freezing, as we talked. I think we agreed this is not ideal but it would require some refactoring to add this at infrastructure level and we don't really expect any additional quantizers to need this in the near future. I would test the quantizers directly, but I didn't see any such existing tests, qat/gptq was the closest I found. My understanding was that we don't want to rewrite/write any new tests for this task. Please correct me if I missed or misunderstood anything.

Regarding adding new tests, I see no problem with that.
I thought of a simple unit test that just initializes some quantizer (can be a test quantizer class not necessarily one of our implemented quantizers), one time with freeze and one time without and just verifies that some quantizer's parameter did or did not change during one or two training steps (not through GPTQ or QAT)

Reply:
will be fixes in #1197